app/javascript/components/ui/Sheet.tsx

```diff
@@ -0,0 +1,98 @@
+import cx from "classnames";
+import * as React from "react";
+
+import { Icon } from "$app/components/Icons";
+
+type SheetProps = {
+  children: React.ReactNode;
+  className?: string;
+  ariaLabel?: string;
+  onClose?: () => void;
+  header?: React.ReactNode;
+};
+
+type SheetHeaderProps = {
+  children: React.ReactNode;
+  className?: string;
+  onClose?: () => void;
+};
+
+type SheetTitleProps = {
+  children: React.ReactNode;
+  className?: string;
+};
+
+type SheetFooterProps = {
+  children: React.ReactNode;
+  className?: string;
+};
+
+type SheetCloseProps = {
```

Comment:
Can we pass through the `React.HTMLAttributes` here instead of listing out the ones we need like `onClick`? Also, can we inline these types or at least co-locate them with their components?